clang/lib/Driver/Driver.cpp

```diff
@@ -5064,8 +5070,17 @@ class OffloadingActionBuilder final {
                 ExtractIRFilesAction, types::TY_Tempfilelist);
 
             ActionList TformInputs{PostLinkAction, TranslateAction};
-            auto *ReplaceFilesAction = C.MakeAction<FileTableTformJobAction>(
-                TformInputs, types::TY_Tempfiletable, types::TY_Tempfiletable);
+            FileTableTformJobAction *ReplaceFilesAction;
+            // Device code files(.spv files) have TY_Filetable type when
+            // -fsycl-dump-device-code option is passed, otherwise
+            // TY_Tempfiletable.
+            if (C.getDriver().isDumpDeviceCodeEnabled())
+              ReplaceFilesAction = C.MakeAction<FileTableTformJobAction>(
+                  TformInputs, types::TY_Filetable, types::TY_Filetable);
+            else
+              ReplaceFilesAction = C.MakeAction<FileTableTformJobAction>(
+                  TformInputs, types::TY_Tempfiletable,
+                  types::TY_Tempfiletable);
```

Comment:
Instead of an if/else construct here, can you populate a `types::ID` variable with the file type needed and use that?